src/coreclr/jit/gentree.cpp

```diff
@@ -8214,6 +8240,7 @@ GenTreeBlk* Compiler::gtNewBlkIndir(ClassLayout* layout, GenTree* addr, GenTreeF
 //
 GenTreeIndir* Compiler::gtNewIndir(var_types typ, GenTree* addr, GenTreeFlags indirFlags)
 {
+    indirFlags |= gtGetFlagsForOperand(GT_IND, addr);
```

Comment:
Why is this needed here? The caller is expected to pass the right `indirFlags`.

Reply:
I bet it's not needed it's just how @MichalPetryka was trying to find the case where it's not set